Makefile

```diff
@@ -1,73 +1,65 @@
 
 .PHONY: ci-build-only
-ci-build-only:
+ci-build-only: ## Kicks off build logic and pulls back deb files.
 	./devops/scripts/ci-build_only.sh
 
 .PHONY: docs-lint
-docs-lint:
+docs-lint: ## Check documentation for common syntax errors.
 # The `-W` option converts warnings to errors.
 # The `-n` option enables "nit-picky" mode.
 	make -C docs/ clean && sphinx-build -Wn docs/ docs/_build/html
 
 .PHONY: docs
-docs:
+docs: ## Build project documentation in live reload for editing
 # Spins up livereload environment for editing; blocks.
 	make -C docs/ clean && sphinx-autobuild docs/ docs/_build/html
 
 .PHONY: flake8
-flake8:
-# Validates PEP8 compliance for Python source files.
+flake8: ## Validates PEP8 compliance for Python source files.
 	flake8 --exclude='config.py' testinfra securedrop-admin \
 		securedrop/*.py securedrop/management \
 		securedrop/tests/functional securedrop/tests/*.py
 
 .PHONY: html-lint
-html-lint:
-# Validates HTML to the best of our ability.
+html-lint: ## Validates HTML in web application template files.
 	html_lint.py --printfilename --disable=optional_tag,extra_whitespace,indentation \
 		securedrop/source_templates/*.html securedrop/journalist_templates/*.html
 
 .PHONY: lint
-# Runs all linting tools (docs, flake8, HTML)
-lint: docs-lint flake8 html-lint
+lint: docs-lint flake8 html-lint ## Runs all linting tools (docs, flake8, HTML).
 
-help:
-	@echo Makefile for developing and testing SecureDrop.
-	@echo Subcommands:
-	@echo "\t docs: Build project documentation in live reload for editing."
-	@echo "\t docs-lint: Check documentation for common syntax errors."
-	@echo "\t flake8: Validates PEP8 compliance for Python source files."
-	@echo "\t html-lint: Validates HTML in web application template files."
-	@echo "\t lint: Runs all linting tools (docs, flake8, HTML)."
-	@echo "\t ci-spinup: Creates AWS EC2 hosts for testing staging environment."
-	@echo "\t ci-teardown: Destroy AWS EC2 hosts for testing staging environment."
-	@echo "\t ci-run: Provisions AWS EC2 hosts for testing staging environment."
-	@echo "\t ci-test: Tests AWS EC2 hosts for testing staging environment."
-	@echo "\t ci-go: Creates, provisions, tests, and destroys AWS EC2 hosts"
-	@echo "\t        for testing staging environment."
-	@echo "\t ci-build-only: Kicks off build logic and pulls back deb files."
-	@echo "\t ci-debug: Prevents automatic destruction of AWS EC2 hosts on error."
+# Explaination of the below shell command should it ever break.
+# 1. Set the field separator to ": ##" and any make targets that might appear between : and ##
+# 2. Use sed-like syntax to remove the make targets
+# 3. Format the split fields into $$1) the target name (in blue) and $$2) the target descrption
+# 4. Pass this file as an arg to awk
+# 5. Sort it alphabetically
+.PHONY: help
+help: ## Print this message and exit.
+	@awk 'BEGIN {FS = ":.*?## "} /^[0-9a-zA-Z_-]+:.*?## / {printf "\033[36m%16s\033[0m : %s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
```

Comment:
This makes the makefile less clear, but it does DRY it up, so I think this is fine. Thanks for adding a comment for clarity 😄  However: this is not a docs only PR and as such we do need full staging CI to run here (especially since CI is using this makefile to run) - can you open this PR from a branch not prefixed with `docs-` so CI can run?